src/txn/txn.c

```diff
@@ -1664,9 +1664,15 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
                 if (!F_ISSET(op, WT_TXN_OP_KEY_REPEATED))
                     WT_ERR(__txn_resolve_prepared_op(session, op, true, &cursor));
 
-                /* Sleep between resolving prepared operations when configured. */
-                __wt_timing_stress(session, WT_TIMING_STRESS_PREPARE_RESOLUTION, NULL);
 #ifdef HAVE_DIAGNOSTIC
+                /*
+                 * Sleep randomly between resolving prepared operations when configured, however,
+                 * avoid causing too much stress when there are a large number of updates.
+                 */
+                if (FLD_ISSET(conn->timing_stress_flags, WT_TIMING_STRESS_PREPARE_RESOLUTION) &&
```

Comment:
Two things: I don't think we need the flag check any more (`__wt_timing_stress` does it, IIRC the check was from an earlier version of the code where we had some more expensive calls in the `if` clause). And secondly, the new code isn't super obvious (why 36? why multiply `i` rather than dividing `txn->mod_count`?) so I think the comment could use a little more detail.

Reply:
Yeah, that's true, the mod is cheap. I'll also add a comment.